buildSrc/build.gradle

```diff
@@ -160,8 +154,8 @@ if (project != rootProject) {
   apply plugin: 'nebula.maven-scm'
 
   allprojects {
-    targetCompatibility = 11
-    sourceCompatibility = 11
+    targetCompatibility = 10
+    sourceCompatibility = 10
```

Comment:
Doesn’t the source/target compatibility need to be 8 not 10, otherwise we can still get into a trouble if an API from JDK 9 or 10 is used?

Reply:
To be quite honest, It's not clear to me why this is 10 or if it needs to be. There are funny interactions here between this and the Groovy plugin as well. This is simply a revert back to what this was before the problematic commit that caused the hadoop builds to begin exploding. I think it's likely we use Java9+ APIs in `.java` code but that isn't used by elasticsearch-hadoop at runtime so all is well.

This is admittedly _very_ awkward and the true solution is to fix the hadoop build to bring it up to Gradle 5.x but we've been kicking that can down the road. This PR is just to unblock the snapshot release builds for the time being. Essentially, this is just the minimal change to bring us back to a known working state.

Reply:
Well, in that case, I think we need to stop kicking the can down the road. I don't like that we are held back from using modern features because of this. I also don't like that we're making a change it seems we don't fully understand (myself included)? I don't think this change is going to prevent us from making this mistake yet again, breaking important downstream builds for days.

Reply:
💯 We had a chat about sorting out the hadoop stuff while back, we just need to prioritize that. I'll put this on the top of my list for when I return from PTO.

Reply:
Thanks @mark-vieira. ❤️